v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/RangeCountDoFn.java

```diff
@@ -56,6 +60,53 @@ final class RangeCountDoFn extends DoFn<Range, Range> implements Serializable {
 
   @JsonIgnore private transient @Nullable DataSource dataSource;
 
+  /**
+   * Various phrases in a SQL exception that indicate timeout. Looking for error code and SQL state
+   * depends on the type of database and the version of driver (for example: <a
+   * href="https://bugs.mysql.com/bug.php?id=96537">bug/96537</a>). The following messages cover for
+   * various database errors like:
+   *
+   * <ul>
+   *   <li>MySQL:
+   *       <ul>
+   *         <li><a
+   *             href=https://dev.mysql.com/doc/mysql-errors/8.4/en/server-error-reference.html#error_er_query_interrupted>error_er_query_interrupted</a>
+   *       </ul>
+   *   <li>PG:
+   *       <ul>
+   *         <li><a
+   *             href=https://www.postgresql.org/docs/current/errcodes-appendix.html>query_canceled</a>
+   *       </ul>
+   *   <li>ORA:
+   *       <ul>
+   *         <li><a href=https://docs.oracle.com/en/error-help/db/ora-01013/?r=23ai>ORA-01013</a>
+   *       </ul>
+   * </ul>
+   */
+  private static final HashSet<String> TIMEOUT_MESSAGES =
+      Sets.newHashSet(
+          "cancel",
+          "cancelled",
+          "abort",
+          "aborted",
+          "interrupt",
+          "interrupted",
+          "terminate",
+          "terminated",
+          "timeout",
+          "time out",
+          "time exceeded");
+
+  private static final Pattern TIMEOUT_MESSAGES_PATTERN =
```

Comment:
There should be a cleaner way to extract the timeout errors here.

Reply:
Thanks. Fixed in the current push.
We had to exchange a list of patterns with a list of error codes, now it's mysql specific which is no doubt a cleaner approach.